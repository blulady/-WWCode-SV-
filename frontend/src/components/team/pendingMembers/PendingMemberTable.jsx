import React from "react";

import TableBody from "./TableBody";
import styles from "./PendingMemberTable.module.css";

const PendingMemberTable = ({ users, ...props }) => {
  return (
    <table className={styles["pending-members-table"]}>
      <thead>
        <tr>
          <th></th>
          <th>Email</th>
          <th>Role</th>
          <th>Status</th>
          <th>Resend Invite</th>
          <th>Actions</th>
        </tr>
      </thead>
      <TableBody
        users={users}
        {...props}
      />
    </table>
  );
};

export default PendingMemberTable;
