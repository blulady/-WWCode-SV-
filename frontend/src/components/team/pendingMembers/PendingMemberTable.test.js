import React from "react";
import { render } from "@testing-library/react";
import PendingMemberTable from "./PendingMemberTable";

const pendingMembers = [{
    email: "abc@example.com",
    role: "volunteer",
    status: "invited"
}];

describe("PendingMemberTable Component Validation Tests", () => {
  test("PendingMemberTable component is rendering", async () => {
      const { container, getByText } = render(<PendingMemberTable users={pendingMembers} />);
      expect(container).toMatchSnapshot();

      expect(getByText("abc@example.com")).toBeTruthy();
      expect(getByText("volunteer")).toBeTruthy();
      expect(getByText("invited")).toBeTruthy();
  });
});
