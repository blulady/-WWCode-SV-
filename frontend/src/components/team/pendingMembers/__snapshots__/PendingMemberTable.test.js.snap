// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingMemberTable Component Validation Tests PendingMemberTable component is rendering 1`] = `
<div>
  <table
    class="pending-members-table"
  >
    <thead>
      <tr>
        <th />
        <th>
          Email
        </th>
        <th>
          Role
        </th>
        <th>
          Status
        </th>
        <th>
          Resend Invite
        </th>
        <th>
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          abc@example.com
        </td>
        <td
          class="wwc-text-capitalize"
        >
          volunteer
        </td>
        <td
          class="wwc-text-capitalize"
        >
          invited
        </td>
        <td>
          <button
            class="invite-button"
            data-bs-toggle="modal"
            type="button"
          >
            Resend Invite
          </button>
        </td>
        <td>
          <TableBody users={props.users} target={props.target}/>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
